engine/security/protect-socket.md

```diff
@@ -1,17 +1,41 @@
 ---
-description: How to setup and run Docker with HTTPS
-keywords: docker, docs, article, example, https, daemon, tls, ca,  certificate
+description: How to setup and run Docker with SSH or HTTPS
+keywords: docker, docs, article, example, ssh, https, daemon, tls, ca,  certificate
 redirect_from:
 - /engine/articles/https/
 - /articles/https/
+- /engine/https/
 title: Protect the Docker daemon socket
 ---
 
 By default, Docker runs through a non-networked UNIX socket. It can also
-optionally communicate using an HTTP socket.
+optionally communicate using SSH or a TLS (HTTPS) socket.
 
-If you need Docker to be reachable through the network in a safe manner, you can
-enable TLS by specifying the `tlsverify` flag and pointing Docker's
+## Use SSH to protect the Docker daemon socket
+
+The `docker` CLI can talk to a remote Docker daemon via SSH,
+by specifying `$DOCKER_HOST` to `ssh://<USERNAME>@<HOST>`.
+
+```console
+$ export DOCKER_HOST=ssh://root@example.com
+$ docker ps
+```
```

Comment:
Perhaps something like this:

> **Note**
>
> The given `USERNAME` must have permissions to access the docker socket on the
> remote machine. Refer to [manage Docker as a non-root user](../../install/linux-postinstall.md#manage-docker-as-a-non-root-user)
> to learn how to give a non-root user access to the docker socket.

The following example creates a [`docker context`](../../context/working-with-contexts.md)
to connect with a remote `dockerd` daemon on `host1.example.com` using SSH, and
as the `docker-user` user on the remote machine:

```console
$ docker context create \
    --docker host=ssh://docker-user@host1.example.com \
    --description="Remote engine" \
    my-remote-engine

my-remote-engine
Successfully created context "my-remote-engine"
```

After creating the context, use `docker context use` to switch the `docker` CLI
to use it, and to connect to the remote engine:

```console
$ docker context use my-remote-engine
my-remote-engine
Current context is now "my-remote-engine"

$ docker info
<prints output of the remote engine>
```

Use use the `default` context to switch back to the default (local) daemon:

```console
$ docker context use default
default
Current context is now "default"
```

Alternatively, use the `DOCKER_HOST` environment variable to temporarily switch
the `docker` CLI to connect to the remote host using SSH. This does not require
creating a context, and can be useful to create an ad-hoc connection with a different
engine:

```console
$ export DOCKER_HOST=ssh://docker-user@host1.example.com
$ docker info
<prints output of the remote engine>
```

Reply:
Markdown for the above (GitHub "suggestions" didn't like the nested code-fences 😞)

    > **Note**
    >
    > The given `USERNAME` must have permissions to access the docker socket on the
    > remote machine. Refer to [manage Docker as a non-root user](../../install/linux-postinstall.md#manage-docker-as-a-non-root-user)
    > to learn how to give a non-root user access to the docker socket.

    The following example creates a [`docker context`](../../context/working-with-contexts.md)
    to connect with a remote `dockerd` daemon on `host1.example.com` using SSH, and
    as the `docker-user` user on the remote machine:

    ```console
    $ docker context create \
        --docker host=ssh://docker-user@host1.example.com \
        --description="Remote engine" \
        my-remote-engine

    my-remote-engine
    Successfully created context "my-remote-engine"
    ```

    After creating the context, use `docker context use` to switch the `docker` CLI
    to use it, and to connect to the remote engine:

    ```console
    $ docker context use my-remote-engine
    my-remote-engine
    Current context is now "my-remote-engine"

    $ docker info
    <prints output of the remote engine>
    ```

    Use use the `default` context to switch back to the default (local) daemon:

    ```console
    $ docker context use default
    default
    Current context is now "default"
    ```

    Alternatively, use the `DOCKER_HOST` environment variable to temporarily switch
    the `docker` CLI to connect to the remote host using SSH. This does not require
    creating a context, and can be useful to create an ad-hoc connection with a different
    engine:

    ```console
    $ export DOCKER_HOST=ssh://docker-user@host1.example.com
    $ docker info
    <prints output of the remote engine>
    ```

Reply:
updated